demo/src/main/java/com/eis/geoCalendar/demo/Behaviour/EventMapBehaviour.java

```diff
@@ -0,0 +1,213 @@
+        this.supportFragmentManager = supportFragmentManager;
+    }
+
+    /**
+     * @param locationRetriever An instance of an object that implements locationRetriever to get the gps position
+     */
+    @Override
+    public void setLocationRetriever(@NonNull LocationRetriever locationRetriever) {
+        this.locationRetriever = locationRetriever;
+    }
+
+    @Override
+    public void setAddEventDialog(@NonNull AbstractAddEventDialog dialog) {
+        this.addEventDialog = dialog;
+    }
+
+    @Override
+    public void setRemoveEventDialog(@NonNull AbstractRemoveEventDialog dialog) {
+        this.removeEventDialog = dialog;
+    }
+
+    /**
+     * Manipulates the map once available.
+     * This callback is triggered when the map is ready to be used.
+     * <p>
+     * If Google Play services is not installed on the device, the user will be prompted to install
+     * it inside the SupportMapFragment. This method will only be triggered once the user has
+     * installed Google Play services and returned to the app.
+     */
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+        //Collect all events from EventManager
+
+        //set onClick listener
+        mMap.setOnMapLongClickListener(this);           //To add Events
+        mMap.setOnInfoWindowLongClickListener(this);    //To delete Events
+
+        //Retrieve current position
+        locationRetriever.setOnLocationAvailableListener(this);
+        locationRetriever.getCurrentLocation();
+
+        if (eventManager != null) {
+            currentEvents = eventManager.getAllEvents();
+            addEventsToMap(currentEvents);
+        }
+    }
+
+    /**
+     * @param position The current available Gps position
+     */
+    @Override
+    public void onLocationAvailable(LatLng position) {
+        moveMap(position);
+    }
+
+    /**
+     * Moves the map's focus to the given position
+     *
+     * @param data The gps position where the map will be focused
+     */
+    private void moveMap(LatLng data) {
+        mMap.moveCamera(CameraUpdateFactory.zoomTo(mMap.getMaxZoomLevel()));
+        mMap.moveCamera(CameraUpdateFactory.newLatLng(data));
+    }
+
+    /**
+     * @param events A bunch of events to position in the map (both description and Position must be defined)
+     */
+    private void addEventsToMap(ArrayList<Event<String>> events) {
+        for (Event<String> event : events) {
+            mMap.addMarker(new MarkerOptions().position(new LatLng(event.getPosition().getLatitude(),
+                    event.getPosition().getLongitude()))
+                    .title(event.getContent()));
+        }
+    }
+
+    /**
+     * Called when user clicks (taps) on the map
+     * Collects the position associated and opens Dialog for creating event
+     *
+     * @param latLng The position on the map where the user clicked
+     */
+    @Override
+    public void onMapLongClick(LatLng latLng) {
```

Comment:
I think there's a contradiction here: the method's name is "onMapLongClick", but the spec says "when the user clicks(tap)".  Maybe it should be made more clear.

Reply:
Well, you should complain with android developers😂
Aside from jokes since this is an Override of GoogleMap, which is a final library class, I can't do anything about its name. I could change the specs accordingly but I think "tap" is the correct verb here since we're dealing with android touch devices

Reply:
To be precise it comes from OnMapReadyCallback interface.

Reply:
Oh sorry, I didn't notice it's an override! 
No problem then, keep all like this.

Reply:
No probs, thank you kindly for the review😉